lib/cws/server.ts

```diff
@@ -38,7 +38,7 @@ export class WebSocketServer extends EventEmitterServer {
   // overload on function from super class
   public on(event: string, listener: Listener): void;
   public on(event: 'error', listener: (err: Error, socket?: Socket) => void): void;
-  public on(event: 'connection', listener: (socket: WebSocket) => void): void;
+  public on(event: 'connection', listener: (socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void): void;
```

Comment:
Can you make `upgradeRequest` optional.

Reply:
I could of course. But the only place where `connection` is emitted does include reference to upgrade request:
```typescript
this.emit('connection', webSocket, this.upgradeReq);
```
And if a user needs to use that upgrade request in the `.on('connection')` callback (e.g., I need query parameters from url), he would have to write something like:
```typescript
// req: IncomingMessage | undefined
wsServer.on('connection', (ws, req) => {
    if (!req) {
        // oops, can't continue
        ws.close();
        
        return;
    }
    
    ws.on('message', msg => { });
    // ...
});
```

Reply:
Oh, yes you are right. Can you instead add additional overload for that function: 

```ts
public on(event: 'connection', listener: (socket: WebSocket) => void): void;
public on(event: 'connection', listener: (socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void): void;
```

Reply:
TSLint is complaining: 
```
This overload and the one on line 41 can be combined into one signature taking `(socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void | (socket: WebSocket) => void`. (unified-signatures)
```
Would you prefer this:
``` typescript
public on(event: 'connection', listener: (socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void): void;
// tslint:disable-next-line: unified-signatures
public on(event: 'connection', listener: (socket: WebSocket) => void): void;
```
this:
``` typescript
public on(event: 'connection', listener: ((socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void) | ((socket: WebSocket) => void)): void;
```
or this :):
``` typescript
public on(
    event: 'connection',
    listener:
      ((socket: WebSocket, upgradeRequest: HTTP.IncomingMessage) => void)
      | ((socket: WebSocket) => void)
  ): void;
```
NB: order of overloads needs to be opposite from what you proposed

Reply:
@trebler to fix above issue can you add  rule to tslint.json ` "unified-signatures": false, ` for all new projects i have this rule in place. Thank you :)

Reply:
@goriunov, all done now: added overload and disabled TSLint `unified-signatures` rule.